hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/scm/proxy/SCMBlockLocationFailoverProxyProvider.java

```diff
@@ -257,8 +249,11 @@ public RetryPolicy getSCMBlockLocationRetryPolicy(String newLeader) {
       @Override
       public RetryAction shouldRetry(Exception e, int retry,
                                      int failover, boolean b) {
-        performFailoverToAssignedLeader(newLeader);
-        return getRetryAction(failover);
+        if (!SCMHAUtils.isRetriableWithNoFailoverException(e)) {
+          performFailoverToAssignedLeader(newLeader);
+        }
+        return SCMHAUtils.getRetryAction(failover, retry, e, maxRetryCount,
```

Comment:
It is weird here that `SCMHAUtils.getRetryAction` does not handle conditions that `SCMHAUtils.isRetriableWithNoFailoverException(e)` is true.